### DISCUSSION TITLE
Welcome to VoxNovel Discussions!

### DISCUSSION THREAD

**Discussion by DrewThomasson** [DrewThomasson/VoxNovel#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Draugvorn** [comment]
Hi Drew,
I tried VoxNovel for the first time yesterday, and I am absolutely amazed by how well it works and the quality of the output! ⭐
I am using a conda environment on WSL (Ubuntu).

Sure, sometimes it hallucinates, but overall, the speech is astonishingly good.
It seems to have trouble with direct speech nested inside narration.

So far, I have just used the default settings.
As so many TTS models are included, do you by any chance have an opinion on the relative performance of the different models?

**Reply by DrewThomasson** [reply]
Actually yes I do!
All models should support voice cloning, and for the ones that don't I've added a script from coqui that uses voice transfer to basically give them voice cloning anyway.


### Voice Models Overview

#### **XTTS_V2 (DEFAULT) Best Transformer based**
- **Performance:** Considered the best among the models for minimal hallucinations.(But still hallucinates as you saw lol)
- **Speed:** Fastest among the transformer models, achieving near real-time audio generation speed on a 4GB GPU, but performs slowly on CPU only.
- **Features:**
  - Supports multiple accents, hence the language dropdown.
  - Produces natural and emotive sounds.
- **Demo:** [Xtts Demo](https://github.com/DrewThomasson/VoxNovel-OLD-/assets/126999465/9e10b36d-b2e9-4462-8bad-13a3d8fce192)
- [Try XTTS V1 for free on hugging face](https://huggingface.co/spaces/coqui/xtts)

#### **Styletts2** Best for fast and ZERO HALLUCINATIONS, (monotone)
- **Description:** Ranks as the second-best model, not transformer-based, instead utilizes e-speak which enhances CPU performance.
- **Speed:** Fast on CPU and compatible with GPU.
- **Features:**
  - Primarily supports English.
  - Output is more monotone; lacks emotional variance.
  - [DEMO](https://styletts2.github.io)
  - [Try it for free on hugging face](https://huggingface.co/spaces/styletts2/styletts2)

#### **Tortoise TTS**
- **Type:** Transformer model.
- **Performance:** Slower than XTTS_V2, consumes more storage, and prone to more disturbing types of hallucinations (e.g., horror movie sounds).
- **Speed:** Faster on GPU but slow on CPU.
- **Features:**
  - Supports only English.
  - Lacks support for multiple accents.
 - **Demo:** Watch a [Tortoise Demo](https://github.com/DrewThomasson/VoxNovel-OLD-/assets/126999465/94e23918-b7e1-4399-935e-179dd12212c3)
 - [Try it for free on hugging face](https://huggingface.co/spaces/Manmay/tortoise-tts)
 
#### **BARK TTS**
- **Description:** Another slow transformer model that generally produces high-quality sound.
- **Issues:**
  - Prone to hallucinating entire words at the end of sentences.
  - May generate background noise and other unintended sounds.
- **Speed:** Performs better on GPU than on CPU.
- **Features:**
  - Only supports English.
  - Limited control over output, leading to potential unpredictability.
  - [Demos can be found on their GitHub](https://github.com/suno-ai/bark)
  - [Try it in hugging face free](https://huggingface.co/spaces/suno/bark)

#### **Other Models (See by clicking the include fast models checkbox in GUI)**
- **Performance:** Very experimental; some models have glitches while others do not.
- **Speed:** Extremely fast on CPU.
- **Features:**
  - Sound similar to Siri, generally lacking in emotional depth.
  - Mostly based on e-speak technology.
  - Voice cloning capability added to all models.

I think that's everything I can think of rn about them

**Comment by senigami** [comment]
hello. Github recommended your project to me as it is very similar to my own. I listened to your sample. it is very nice. I starred your project and I'm going to scan your code to see how it compares to my project. you are welcome to have a look at my repo as well.  Here is the demo page for it. https://senigami.github.io/audiobook-studio/
